pkg/client/unit.go

```diff
@@ -251,25 +252,37 @@ func (u *Unit) RegisterDiagnosticHook(name string, description string, filename
 }
 
 // updateConfig updates the configuration for this unit, triggering the delegate function if set.
-func (u *Unit) updateState(exp UnitState, logLevel UnitLogLevel, cfg *proto.UnitExpectedConfig, cfgIdx uint64) bool {
+func (u *Unit) updateState(exp UnitState,
+	logLevel UnitLogLevel,
+	features *proto.Features,
+	cfg *proto.UnitExpectedConfig,
+	cfgIdx uint64) bool {
+
 	u.expMu.Lock()
 	defer u.expMu.Unlock()
 	changed := false
+
 	if u.exp != exp {
 		u.exp = exp
 		changed = true
 	}
+
 	if u.logLevel != logLevel {
 		u.logLevel = logLevel
 		changed = true
 	}
+
 	if u.configIdx != cfgIdx {
 		u.configIdx = cfgIdx
-		if !gproto.Equal(u.config.GetSource(), cfg.GetSource()) {
+		if !gproto.Equal(u.config.GetSource(), cfg.GetSource()) ||
+			!gproto.Equal(u.features, features) {
```

Comment:
I think this will be a source of bugs, because we will now send unchanged unit configurations to clients of this code and they will need to filter them out.

The intent of the elastic agent client is to only send unit modified events to a client of the package when the actual unit configuration changed. This breaks that contract, and will send unchanged unit configurations if only the feature flags have been modified.